lib/RestructureCFG/InlineDispatcherSwitch.cpp

```diff
@@ -509,6 +509,65 @@ static void processNestedWeavedSwitches(SwitchNode *Switch) {
   }
 }
 
+static RecursiveCoroutine<bool> containsNonLocalCF(ASTNode *Node) {
+  switch (Node->getKind()) {
+  case ASTNode::NK_List: {
+    SequenceNode *Seq = llvm::cast<SequenceNode>(Node);
+
+    bool HasNonLocalCF = false;
+    for (ASTNode *N : Seq->nodes()) {
+      HasNonLocalCF = HasNonLocalCF or rc_recur containsNonLocalCF(N);
+    }
+
+    rc_return HasNonLocalCF;
+  } break;
+  case ASTNode::NK_Scs: {
+
+    // We do not want to move around a `ScsNode` for any reason
+    rc_return false;
+  } break;
+  case ASTNode::NK_If: {
+    IfNode *If = llvm::cast<IfNode>(Node);
+
+    bool HasNonLocalCF = false;
+    if (If->hasThen()) {
+      ASTNode *Then = If->getThen();
+      HasNonLocalCF = HasNonLocalCF or rc_recur containsNonLocalCF(Then);
+    }
+    if (If->hasElse()) {
+      ASTNode *Else = If->getElse();
+      HasNonLocalCF = HasNonLocalCF or rc_recur containsNonLocalCF(Else);
+    }
```

Comment:
Change this code so that it early returns `true` when possible.

Like, if the `then` has non-local control flow, there's no need to look inside the `else`.

Reply:
Right, applied the suggested changes [here](https://github.com/revng/revng-c/blob/f15b7b1a9506a37ed6d348f7dd6d880b1d93df16/lib/RestructureCFG/InlineDispatcherSwitch.cpp#L539)